src/NServiceBus.AcceptanceTests/HandlerContext/When_saga_with_old_and_new_api_is_created.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace NServiceBus.AcceptanceTests.Sagas
+{
+    using System;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.Saga;
+    using NUnit.Framework;
+
+    public class When_saga_with_old_and_new_api_is_created : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public void Should_process_the_saga_successfully()
+        {
+            var dataId = Guid.NewGuid();
+
+            Scenario.Define<Context>()
+                .WithEndpoint<SagaEndpoint>(b =>
+                    b.Given(bus =>
+                    {
+                        bus.SendLocal(new StartSaga
+                                       {
+                                           DataId = dataId
+                                       });
+                    })
+                    .When(c => c.SagaStarted, (bus, c) => { bus.Publish<SomethingHappenedEvent>(m => m.DataId = dataId); })
+                )
+                .Done(c => c.SagaCompleted)
+                .Run(TimeSpan.FromSeconds(10));
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool SagaCompleted { get; set; }
+            public bool SagaStarted { get; set; }
+        }
+
+        public class SagaEndpoint : EndpointConfigurationBuilder
+        {
+            public SagaEndpoint()
+            {
+                EndpointSetup<DefaultPublisher>();
+            }
+
+            public class Saga1 : Saga<Saga1.Saga1Data>, // Daniel: I think we need a new base class which doesn't have the IBus already predefined or should we use the same base and advice users to use the context information?
+                IAmStartedByMessages<StartSaga>, // Daniel: I think we shouldn't allow this. But technically it is possible
+                IAmStartedByMessage<StartSaga>, // Daniel: see above you need to pick and choose
+                IAmStartedByEvent<SomethingHappenedEvent>, // Daniel: I'm started by event?
```

Comment:
Or turn it around. Because in practice there is rarely a case where a
message should not start a saga why not make it by default so that a
message can start a saga and you have to define which messages do not start
a saga?